p2panda-rs/src/document/document.rs

```diff
@@ -133,22 +133,37 @@ impl Document {
     }
 }
 
-/// A struct for building documents from a collection of operations. When calling `build()
-/// a document is returned wrapped in a result. The build will error if the operations passed
-/// don't follow documents validation criteria.
+/// A struct for building [documents][`Document`] from a collection of [operations with metadata][`crate::operation::OperationWithMeta`].
 ///
-/// Validation checks the following:
-/// - There should be exactly one CREATE operation.
-/// - All operations should be causally connected to the root operation.
-/// - All operations should follow the same schema.
-/// - No cycles exist in the graph.
+/// ## Example
+///
+/// ```
+/// # extern crate p2panda_rs;
+/// # #[cfg(test)]
+/// # mod tests {
+/// # use rstest::rstest;
+/// # use p2panda_rs::document::DocumentBuilder;
+/// # use p2panda_rs::operation::OperationWithMeta;
+/// # use p2panda_rs::test_utils::meta_operation;
+/// #
+/// # #[rstest]
+/// # fn main(meta_operation: OperationWithMeta) -> () {
+/// // You need a `Vec<OperationWithMeta>` that includes the `CREATE` operation
+/// let operations: Vec<OperationWithMeta> = vec![meta_operation];
```

Comment:
Not sure about the name `meta_operation` suggests to me a special kind of operation which is meta, rather than a normal operation with meta data attached (I know the whole naming for this thingy is TBC but in the context of this PR..... that's my feeling ;-p)

Reply:
You mean instead have `operation_with_meta` as the variable name? I could also rename the whole fixture.

Reply:
Yeh, either that, or just have it as `operation`

Reply:
Nice! +1 for renaming the variable from `meta_operation` to `operation` in the example (not the fixture). Otherwise nice example @cafca, happy to merge it after that small change.